pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -560,7 +595,7 @@ func (nc *NodeController) MonitorNodeStatus() error {
 				nc.deletePods(node.Name)
 			}
 			if lastReadyCondition.Status == api.ConditionUnknown &&
-				nc.now().After(lastReadyCondition.LastProbeTime.Add(nc.podEvictionTimeout-gracePeriod)) {
+				nc.now().After(nc.nodeStatusMap[node.Name].timestamp.Add(nc.podEvictionTimeout-gracePeriod)) {
```

Comment:
Can't we just store in the map now-node.Status.LastProbeTime (updated every time this function runs), and then say that is an upper bound on the clock skew?